### PR TITLE
ref(replays): Refactor trace data loading on Replay Details page

### PR DESCRIPTION
**Pull request by ryan953**
Incremental loading of trace data.

What this change does is move loading trace data into a react context provider, so the data can be loaded outside of the Trace tab itself. This allows us to load the data once per pageload, not every time the trace tab is opened!

Also, this change iterates over all pages of data to fetch every traceId that's related to the replay. It could be many ids, and once the loading process is started it'll eventually complete as long as we stay on the page. A follow up change could, for example, start loading the trace data before the trace tab is clicked, so it would render quicker once the user wants to see it. 

Initially I had a loading indicator/counter that would show how many traces were expected, and how many were loaded already. I pulled this out
a) Because the PR is slightly simpler without it (still have the `detailsRequests` state, etc) 
b) Turns out that we can load the first page of 50 traces quick enough to fill up the screen. Then once the screen is filled up it looks ready to use, and you an actually start clicking into the traces that are rendered! We could followup with a subtle loader, maybe loading-rows injected into the table, or a single spinner at the bottom. To be designed.

Using https://sentry.dev.getsentry.net:7999/replays/javascript:449205bb6d0b47bfaf1e6407febda9ec/ as an example replay, here are some screen shots of the process:

| Desc | Image |
| --- | --- |
| Details Loading | loading - with border |
| Incremental Loading 1 | loading - partial 1 |
| Incremental Loading 2 | loading - partial data |
| Loading Complete | loading - all data |

Fixes https://github.com/getsentry/team-replay/issues/60

### CODE REVIEW

**Review by billyvg — COMMENTED** [review]
It currently flashes an error message when switching from Trace to Trace2 on a fresh pageload (and goes away after rows start rendering).

**Comment by ryan953** [comment]
> It currently flashes an error message when switching from Trace to Trace2 on a fresh pageload (and goes away after rows start rendering).

Fixed!

File: static/app/views/replays/detail/trace/index.tsx [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
-function PerfDisabled() {
-  return (
-    <FeatureDisabled
-      featureName={t('Performance Monitoring')}
-      features={features}
-      hideHelpToggle
-      message={t('Requires performance monitoring.')}
-    />
-  );
-}
```
Comment (billyvg) [inline static/app/views/replays/detail/trace/index.tsx:24]: Why get rid of the feature flag?

File: static/app/views/replays/detail/trace/index.tsx [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
   );
 }
 
-export default TraceFeature;
+const LoadingCount = styled('div')`
+  position: absolute;
+  bottom: 0;
+  right: 0;
+  background: red;
+`;
```
Comment (billyvg) [inline static/app/views/replays/detail/trace/index.tsx:52]: Are we planning on keeping this?

File: static/app/views/replays/detail/trace/replayTransactionContext.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      detailsComplete: false,
+      traces: [],
+    });
+
+    let cursor = {
+      cursor: '0:0:0',
+      results: true,
+      href: '',
+    } as ParsedHeader;
+    while (cursor.results) {
```
Comment (billyvg) [inline static/app/views/replays/detail/trace/replayTransactionContext.tsx:161]: A bit concerned about the # of requests for enterprise orgs (could be 400+ requests)

**Review by billyvg — COMMENTED** [review]
Does context get cleared when we switch to a different view (e.g. Issues) or to a different replay?

File: static/app/views/replays/detail/trace/index.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/views/replays/detail/trace/index.tsx:24]: oops. i will bring that back

File: static/app/views/replays/detail/trace/index.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/replays/detail/trace/index.tsx:52]: ahh, this should've come out. lemme re-push

File: static/app/views/replays/detail/trace/replayTransactionContext.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/replays/detail/trace/replayTransactionContext.tsx:161]: I agree.
It's going to break right away, and if we wait for _after_ ingest adds a bias for replays then it'll break more often and we will be catching up trying to fix it in Q3. 

IMO we should be re-building this tab almost from scratch in Q2. It needs to be searchable and filterable so you can grok through hundreds of traces, and load much less data in the process. That's going to need new api calls probably, etc.

**Review by billyvg — APPROVED** [review]

**Pull request merged**